pkg/plugins/manager/loader/initializer/initializer.go

```diff
@@ -106,5 +126,23 @@ func getPluginSettings(pluginID string, cfg *config.Cfg) pluginSettings {
 		ps[k] = v
 	}
 
-	return ps
+	return ps, nil
+}
+
+func (i *Initializer) oauth2OnBehalfOfVars(pluginID string, oauthAppInfo *oauthserver.ExternalServiceRegistration) ([]string, error) {
+	cli, err := i.oauthServer.SaveExternalService(context.Background(), &oauthserver.ExternalServiceRegistration{
+		ExternalServiceName:    pluginID,
+		Permissions:            oauthAppInfo.Permissions,
+		ImpersonatePermissions: oauthAppInfo.ImpersonatePermissions,
+		Key:                    oauthAppInfo.Key,
+	})
+	if err != nil {
+		return nil, err
```

Comment:
I don't think we have great coverage for when the initialize step of plugin loading fails, so it would be good to add a test here that covers this case (in loader_test.go). Right now any error will cause the loader to completely error out, but I have the feeling that behaviour should change to probably log and skip instead

Reply:
Added the test but I don't think the initializer should "ignore" the error (just log it and skip) since this is initializing just one plugin. It should be the [`loader`](https://github.com/grafana/grafana/blob/main/pkg/plugins/manager/loader/loader.go#L165) the one to decide if ignore one plugin that failed to initialize (and that's out of the scope of this PR)

Reply:
>I don't think the initializer should "ignore" the error (just log it and skip) since this is initializing just one plugin. It should be the [loader](https://github.com/grafana/grafana/blob/main/pkg/plugins/manager/loader/loader.go#L165) the one to decide if ignore one plugin that failed to initialize (and that's out of the scope of this PR)

Yep that's what I had meant originally. Fair point about being out of scope, but I'd like to patch that before we go live with this feature 👍

Reply:
can do that as a separate PR (targeting `main`) :+1:

Reply:
https://github.com/grafana/grafana/pull/66210